src/frontend/Account.ts

```diff
@@ -142,16 +125,19 @@ class Account {
    */
   async disconnect(walletType: AllowedWallet): Promise<void> {
     if (walletType === "magic" && this.magic?.user) await this.magic.user.logout();
-    if (this.walletConnectProvider?.disconnect) this.walletConnectProvider.disconnect();
+    if (walletType === "walletConnect" && this.walletProvider) {
+      const walletConnectProvider = this.walletProvider.provider as WalletConnectProvider;
+      await walletConnectProvider.disconnect();
+    }
```

Comment:
Can you leave a comment explaining why this is necessary in the case of walletconnect?